runtime/src/main/java/org/corfudb/protocols/wireprotocol/TokenRequest.java

```diff
@@ -68,7 +70,9 @@ public TokenRequest(Long numTokens, Set<UUID> streams,
     public TokenRequest(Long numTokens, Set<UUID> streams) {
         if (numTokens == 0) {
             this.reqType = TK_QUERY;
-        } else if (streams == null || streams.size() == 0) {
+        } else if (numTokens == Integer.MAX_VALUE) {
```

Comment:
This seems to be incorrect and one of the reasons the PR had an issue in the first place. Just add a new constructor which takes the request type, and reject if the input is anything but TK_QUERY_ALL. Set numtokens to 0.